sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -169,6 +169,26 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
     checkAnswer(
       testData.filter("key > 90"),
       testData.collect().filter(_.getInt(0) > 90).toSeq)
+
+    checkAnswer(
+      testData.filter("key > 9.0e1"),
+      testData.collect().filter(_.getInt(0) > 90).toSeq)
+
+    checkAnswer(
+      testData.filter("key > .9e+2"),
+      testData.collect().filter(_.getInt(0) > 90).toSeq)
+
+    checkAnswer(
+      testData.filter("key > 0.9e+2"),
+      testData.collect().filter(_.getInt(0) > 90).toSeq)
+
+    checkAnswer(
+      testData.filter("key > 900e-1"),
+      testData.collect().filter(_.getInt(0) > 90).toSeq)
+
+    checkAnswer(
+      testData.filter("key > 900.0E-1"),
+      testData.collect().filter(_.getInt(0) > 90).toSeq)
```

Comment:
Could you please add a case for literals like `9.e+1`? This should be accepted according to the defined parser rules.